cpp/arcticdb/version/version_map_batch_methods.hpp

```diff
@@ -117,6 +120,130 @@ inline std::shared_ptr<std::unordered_map<std::pair<StreamId, VersionId>, AtomKe
     return output;
 }
 
+struct StreamVersionData {
+    explicit StreamVersionData(const pipelines::VersionQuery& version_query) {
+        react(version_query);
+    }
+
+    StreamVersionData() = default;
+
+    size_t count_ = 0;
+    LoadParameter load_param_ = LoadParameter{LoadType::LOAD_LATEST_UNDELETED};
+
+    void react(const pipelines::VersionQuery& version_query) {
+        util::variant_match(version_query.content_, [that = this](const auto &query) {
+            that->do_react(query);
+        });
+    }
+
+    void do_react(std::monostate) {
+        ++count_;
+    }
+
+    void do_react(const pipelines::SpecificVersionQuery& specific_version) {
+        ++count_;
+        switch(load_param_.load_type_) {
+        case LoadType::LOAD_LATEST_UNDELETED:
+            load_param_ = LoadParameter{LoadType::LOAD_DOWNTO, specific_version.version_id_};
+            break;
+        case LoadType::LOAD_DOWNTO:
+            util::check(load_param_.load_until_.has_value(), "Expect LOAD_DOWNTO to have version specificed");
+            load_param_.load_until_ = std::min(load_param_.load_until_.value(), specific_version.version_id_);
+            break;
+        case LoadType::LOAD_FROM_TIME:
```

Comment:
@vasil-pashov moving the comment here https://github.com/man-group/ArcticDB/pull/423/files#r1205057639

Reply:
If we add it, I think that if we add it, we should do the same for all the switch/cases, to be consistent. So far, I don't see any place in the code base where it is used and on my opinion that's just used to avoid potential warnings/user clarification. I don't see a big benefit.

Reply:
It's kind of nice, and we do use [[nodiscard]] etc elsewhere. The standard versions of these weren't available reliably when the project was started, so it's not everywhere. Likewise there are some preprocessor-defined GCC attributes that are now part of the standard, that we should replace at some point